jsapp/js/components/usageLimits/usageCalculations.ts

```diff
@@ -0,0 +1,246 @@
+          truncate(data.total_nlp_usage['asr_seconds_current_year'] / 60)
+        ),
+        monthlyTranslationChars:
+          data.total_nlp_usage['mt_characters_current_month'],
+        yearlyTranslationChars:
+          data.total_nlp_usage['mt_characters_current_year'],
+      });
+    });
+  }, []);
+
+  function setLimitThresholds(limitName: string, limitValue: string | number) {
+    // If user is subscribed to a plan assign limit for that plan
+    if (limitValue === '') {
+      switch (limitName) {
+        case 'storage':
+          limitValue =
+            state.subscribedProduct?.[0].items[0].price.product.metadata
+              .storage_bytes_limit;
+          break;
+        case 'data':
+          limitValue =
+            state.subscribedProduct?.[0].items[0].price.product.metadata
+              .submission_limit;
+          break;
+        case 'translation_char':
+          limitValue =
+            state.subscribedProduct?.[0].items[0].price.product.metadata
+              .nlp_character_limit;
+          break;
+        case 'transcription_minutes':
+          limitValue =
+            state.subscribedProduct?.[0].items[0].price.product.metadata
+              .nlp_seconds_limit;
+          break;
+        default:
+          break;
+      }
+    }
+    switch (limitName) {
+      case 'storage':
+        setSubscribedStorageLimit(limitValue);
+        break;
+      case 'data':
+        setSubscribedSubmissionLimit(limitValue);
+        break;
+      case 'translation_char':
+        setTranslationChars(limitValue);
+        break;
+      case 'ttranscription_minutes':
+        setTranscriptionMinutes(limitValue);
+        break;
+      default:
+        break;
+    }
+  }
+
+  // Check Thresholds
+  useMemo(() => {
+    const thresholds = envStore.data.free_tier_thresholds;
+    Object.entries(thresholds).forEach((entry) => {
+      const [key, value] = entry;
+      if (value === null) {
+        setLimitThresholds(key, '');
+        return;
+      } else if (value > 0) {
+        setLimitThresholds(key, value);
+      } else if (value <= 0) {
+        setLimitThresholds(key, 'unlimted');
+      }
+    });
+  }, []);
+
+  // Check if usage is more than limit
+  useMemo(() => {
+    // Check yearly vs monthly / community
+    let interval;
+    if (state.subscribedProduct?.length > 0) {
+      interval = state.subscribedProduct?.[0].items[0].price.recurring.interval;
+    }
+    exceedList = [];
+
+    if (subscribedStorageLimit && typeof subscribedStorageLimit !== 'string') {
+      if (usage.storage > subscribedStorageLimit) {
+        exceedList.push(t('storage'));
+      }
+    }
+    // If subscribed plan is month or community plan
+    if (interval === 'month' || interval === undefined) {
+      if (
+        subscribedSubmissionLimit &&
+        typeof subscribedSubmissionLimit !== 'string'
+      ) {
+        if (usage.monthlySubmissions > subscribedSubmissionLimit) {
```

Comment:
A version of this check happens 6 times in this block; it might read a bit easier if you collapsed each one into something like this rough example:

```code
if (isOverLimit(usage.monthlySubmissions, subscribedSubmissionList)) {
  exceedList.push(t('submissions'));
}
```